helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/ReadOnlyWagedRebalancer.java

```diff
@@ -69,6 +71,10 @@ public void updateChangeDetectorSnapshots(ResourceControllerDataProvider dataPro
       super(new ZkBucketDataAccessor(metadataStoreAddress), clusterName);
     }
 
+    ReadOnlyAssignmentMetadataStore(ZkBucketDataAccessor zkBucketDataAccessor, String clusterName) {
```

Comment:
Optional, but if there is no other use cases, please just remove the older constructor. The newer one is more specific and avoid misusage, IMO.